src/core/nodeOps.ts

```diff
@@ -180,15 +182,41 @@ export const nodeOps: RendererOptions<TresObject, TresObject | null> = {
       disposeMaterialsAndGeometries(node)
       deregisterCameraIfRequired(node as Object3D)
       deregisterAtPointerEventHandlerIfRequired?.(node as TresObject)
+      invalidateInstance(node as TresObject)
+      node.dispose?.()
+      
     }
 
-    invalidateInstance(node as TresObject)
-    node.dispose?.()
   },
-  patchProp(node, prop, _prevValue, nextValue) {
+  patchProp(node, prop, prevValue, nextValue) {
     if (node) {
       let root = node
       let key = prop
+      if (node.__tres.primitive && key === 'object' && prevValue !== null) {
+        // If the prop 'object' is changed, we need to re-instance the object and swap the old one with the new one
+        const newInstance = nodeOps.createElement('primitive', undefined, undefined, { 
```

Comment:
It seems like this doesn't break references in `<script>` per se. The `:object`s still exist. But they're copied. That means something like the following (probably) wouldn't have the user's desired effect in the playground example:

```
useRenderLoop().onLoop(({elapsed}) => {
  torus.position.y = Math.sin(elapsed)
})
```

Whenever the primitive object becomes `torus`, `torus` is copied, along with its attributes. But the copy doesn't move like `torus` moves, because it's a different `Torus`.

Reply:
@andretchen0 we will need to explicitly mention on the docs that the user needs to animate the `ref` (primitiveRef) but not `the object` (torus) following the same DX as they were using a `<TorusMesh ref="torusRef" />` 

https://github.com/Tresjs/tres/assets/4699008/650d4633-c394-4a41-8c63-0b668cd878c5

There are several reasons why we need to `clone` the object:
- Since we soft-swipe the instances when `:object` changes (we maintain the reference but copy all the attributes) that would modify the original object from the setup leading to unwanted side effects.
It is not possible to pass a `ref` or `reactive` as @JaimeTorrealba mentioned in the thread. With clone, we can ensure that swiping between models is possible.

This is what happens without the `clone` 

https://github.com/Tresjs/tres/assets/4699008/b7ce86cc-44a1-4985-a91a-94b81df9c256

It's either has this limitation of having to use the ref coming from the custom renderer to animate or `:object` can't be reactive and users will need to use `v-if` and `v-else` to swipe models:

```
<primitive v-if="showA" :object="modelA" />
<primitive v-else :object="modelB" />
```

I balance through the first option.

Reply:
@alvarosabu 

I think the PR as it currently stands is an improvement. I don't want to stand in the way of improvements.

Reply:
This sort of error is an itch I can't stop scratching, so if you want to keep scratching ...

> Since we soft-swipe the instances when :object changes (we maintain the reference but copy all the attributes) that would modify the original object from the setup leading to unwanted side effects.

Can you elaborate on unwanted side effects?

Technical issues aside, there's a point that isn't clear for me conceptually. To illustrate:

```vue
<script>
...
const userMesh = new Mesh(...)
userMesh.position.x = 100

const userMesh2 = new Mesh(...)
userMesh2.position.x = 200
</script>

<template>
...
<primitive ref="primitiveRef" :object="userMesh" :position-x="300" />
...
</template>
```

* Should `userMesh`'s x be 300? (300 is the value in the `primitive`)
* And later, if the user does `primitiveRef.value.object = userMesh2`, does that set `userMesh2`'s x to 300?

For R3F, the answer to both is "yes", it seems. [Here's a StackBlitz.](https://stackblitz.com/edit/vitejs-vite-ggmpvs?file=src%2Fcomponents%2FPrimitiveTest.tsx) Notice that the rotation of the TorusRing increments with every frame, but resets whenever it's made the `primitive.object`. And the Torus would normally be placed at x = 2000, but its x is set to the primitive's value.

That seems like the right solution to me.

Reply:
https://github.com/Tresjs/tres/blob/8408351e870a3e3cd530abd5b7ca58359c38f3e2/src/core/nodeOps.ts#L197

I understand that this will make a clone, which is required at the moment, but is there a reason why it's necessary to go through `nodeOps.createElement`? Is it updating something on the Vue side, maybe?

Reply:
It seems like we're running into the "is-a" vs. "has-a" debate with Tres, though JS-style instead of classical OOP.

As it stands, the `node` in `nodeOps.ts` **is a** `TresObject`. Most of the time, it's a `THREE.Object3D` with some extra fields for Vue/Tres data. 

It's usually the case that `has-a` relationships have more flexibility. So, maybe we try that here with something like this shape for our node?

```
{
    tres: {
        // our Tres data
    },
    three: {
        // whatever THREE.Object3D, etc. is connected with this node – or none!
    }
}
```

Because `node` just has an optional reference to a `THREE` object, this has a nice knock-on effect in that we could have `<primitive />` as a placeholder without an `:object`.

I'll give it a shot and see how many errors that makes. ;)